src/game_battlealgorithm.cpp

```diff
@@ -1549,15 +1555,15 @@ bool Game_BattleAlgorithm::Item::Execute() {
 			this->sp = std::max<int>(0, std::min<int>(item.recover_sp_rate * GetTarget()->GetMaxSp() / 100 + item.recover_sp, GetTarget()->GetMaxSp() - GetTarget()->GetSp()));
 		}
 
-		// We allocate a copy of the target in order to check if we can apply states..
-		auto target_copy = target->Clone();
+		// Make a copy of the target's state set and see what we can apply.
+		auto target_states = target->GetStates();
```

Comment:
> Remove Game_Battler::Clone() 

How does this copy? GetStates returns a reference.

Reply:
I say `auto x = ...` not `auto& x =` I am making a copy locally.

Reply:
ah, don't know how auto deduces the type, thought it just takes the return type.

Reply:
`auto` drops references. If you use `decltype(auto)` it will retain references.